DEVDOC.md

```diff
@@ -313,3 +313,60 @@ A few useful homemade packages:
 ### Java
 
 > TODO
+
+## Code generated at compile time
+
+A big amount of code is generated at compile time. Code is typically generated
+via APIC (API Compiler, `cmd/apic/`). APIC takes as input `.api` and `.tmpl`
+files, and it generates code in another language like Go, C++ or Java.
+
+```
+
+               +----------+
+  foo.api  --->|          |
+               |   APIC   |---> foo.go (or foo.cpp, or foo.java, ...)
+  foo.tmpl --->|          |
+               +----------+
+
+```
+
+`.api` files are gapil ("graphics API language") sources. Gapil is a
+domain-specific language to specify a graphics API, it is documented in
+`gapil/README.md`. AGI currently supports only Vulkan, but its ancestor GAPID
+was designed to support arbitrary graphics API. Vulkan is described by API files
+located under `gapis/api/vulkan/`, the top-level file is
+`gapis/api/vulkan/vulkan.api`.
+
+`.tmpl` files are template files. The templating language is documented in
+`gapis/api/templates/README.md`. There are various template files in the project
+to generate Go, C++ and Java code.
+
+APIC is a gapil compiler, its wrapper command is defined in `cmd/apic/`, but the
+actual compiler logic is defined in `gapil/`. In a nutshell, APIC parses the
+`.api` files to gather information about the graphics API, and then instanciates
+the templates in the `.tmpl` files to generate code.
+
+### Where does the generated code ends up?
+
+The generated code is not checked under version control. It is generated at
+compilation time by Bazel rules calling APIC, resulting in files that can be
+seen under e.g. `bazel-bin/`.
+
+For instance, generated C++ code related to the interceptor spy layer can be
+found in:
+
+```
+user@machine:~/work/agi$ find -L bazel-bin -name '*_spy_*.cpp'
+bazel-bin/gapii/cc/vulkan_spy_0.cpp
+bazel-bin/gapii/cc/vulkan_spy_subroutines_1.cpp
+bazel-bin/gapii/cc/vulkan_spy_subroutines_0.cpp
+bazel-bin/gapii/cc/vulkan_spy_2.cpp
+bazel-bin/gapii/cc/vulkan_spy_3.cpp
+bazel-bin/gapii/cc/vulkan_spy_helpers.cpp
+bazel-bin/gapii/cc/vulkan_spy_1.cpp
+```
+
+This is one of the reason why it's good to use the `cmd-gofuse` setup: it
+enables developers to see generated code where it would logically reside for
+compilation, under the `<agi-gofuse>` directory, alongside the code that is
+under version control.
```

Comment:
```suggestion
To get a unified file tree view, interleaving source files with generated files, you
can use the `cmd-gofuse` setup. `cmd-gofuse` will link the generated source files
in the same directories as their siblings in the same packages/namespaces.
```

I don't think it needs to be explained that generated files are not checked in, and this uses a bit of a more objective tone.

Reply:
done